File: src/groups.rs
/// Create new group.
#[post("/groups")]
pub fn create_group() -> &'static str {
    "New group was created."
}

/// Edit group.
#[put("/groups/<group_id>")]
pub fn edit_group(group_id: u64) -> &'static str {
    "Group was saved."
}

/// Delete group.
#[delete("/groups/<group_id>")]
pub fn delete_group(group_id: u64) -> &'static str {
    "Group was deleted."
}

File: backend/groups.go
package backend

import (
	"gorm.io/gorm"
)

type Group struct {
	ID    uint64 `gorm:"primaryKey"`
	Name  string `gorm:"unique,notNull"`
	Links []Link `gorm:"constraint:OnDelete:CASCADE;"`
}

func GetGroups(db *gorm.DB) ([]Group, error) {
	var groups []Group
	result := db.Model(&Group{}).Preload("Links").Find(&groups)
	if result.Error != nil {
		return nil, result.Error
	}
	return groups, nil
}

func CreateGroup(db *gorm.DB, groupName string) (Group, error) {
	group := Group{
		Name: groupName,
	}
	result := db.Create(&group)
	if result.Error != nil {
		return Group{}, result.Error
	}

	return group, nil
}

func UpdateGroup(db *gorm.DB, id uint64, groupName string) (Group, error) {
	var group Group
	db.First(&group, id)

	group.Name = groupName
	result := db.Save(&group)
	if result.Error != nil {
		return Group{}, result.Error
	}

	return group, nil
}

func DeleteGroup(db *gorm.DB, id uint64) error {
	result := db.Delete(&Group{}, id)
	if result.Error != nil {
		return result.Error
	}
	return nil
}
